cmd/crc-embedder/cmd/embed.go

```diff
@@ -124,16 +135,44 @@ func getEmbedFileList(goos string, destDir string) []string {
 	return fileList
 }
 
-func downloadDataFiles(goos string, destDir string) ([]string, error) {
+func getAllComponentNames(goos string) []string {
+	var components []string
+	for component := range dataFileUrls[goos] {
+		components = append(components, component)
+	}
+	return components
+}
+
+func shouldDownload(components []string, component string) bool {
+	if len(components) == 0 {
+		return true
+	}
+	for _, v := range components {
+		if v == component {
+			return true
+		}
+	}
+	return false
+}
+
+func downloadDataFiles(goos string, components []string, destDir string) ([]string, error) {
 	downloadedFiles := []string{}
 	downloads := dataFileUrls[goos]
-	for _, dl := range downloads {
+
+	for componentName, dl := range downloads {
+		if !shouldDownload(components, componentName) {
+			continue
+		}
 		filename, err := download.Download(dl.url, destDir, dl.permissions, nil)
 		if err != nil {
 			return nil, err
 		}
 		downloadedFiles = append(downloadedFiles, filename)
 	}
 
+	if len(downloadedFiles) == 0 {
```

Comment:
This could even be stricter, such as `if len(components) != 0 && len(components) != len(downloadedFiles)`